agent/api/task.go

```diff
@@ -423,6 +450,27 @@ func (task *Task) dockerHostConfig(container *Container, dockerContainerMap map[
 		}
 	}
 
+	// Set network mode in host config. Since we create internal containers for
+	// volumes and setting up networking for ENIs, we handle those special cases
+	// via a switch statement
+	switch container.Name {
```

Comment:
Same question as above.  I worry that depending on the name (and string comparisons in general) can be fragile.